features/client/http_proxy_auth/feature.ts

```diff
@@ -0,0 +1,43 @@
+import assert from 'assert';
+import { randomUUID } from 'crypto';
+import { Client, Connection } from '@temporalio/client';
+import { Feature } from '@temporalio/harness';
+
+export async function workflow(): Promise<void> {
+  return;
+}
+
+export const feature = new Feature({
+  workflow,
+  async alternateRun(runner) {
+    assert(runner.options.proxyUrl);
+
+    const connection = await Connection.connect({
+      ...runner.connectionOpts,
+      proxy: {
```

Comment:
Per the comment at https://github.com/temporalio/sdk-typescript/pull/1411#discussion_r1593057635, I am afraid this has side effects that affect other features. If environment variables are already supported and that's the primary way to configure, that's good enough, I don't think we need per-client config then (and you'll have to change your test to do what Go did which is make a sub process).